Mathlib/Data/Finset/Card.lean

```diff
@@ -656,6 +657,27 @@ theorem card_eq_three [DecidableEq α] :
       or_self_iff, card_singleton]
 #align finset.card_eq_three Finset.card_eq_three
 
+lemma covby_tfae :
+    List.TFAE [t ⋖ s, t ⊆ s ∧ (s \ t).card = 1, ∃ a, a ∉ t ∧ insert a t = s,
+      ∃ a, a ∈ s ∧ t = s.erase a] := by
+  tfae_have 1 ↔ 3
+  · exact covby_iff_exists_insert
+  tfae_have 1 ↔ 4
+  · exact covby_iff_exists_erase
+  tfae_have 3 → 2
+  · rintro ⟨a, ha, rfl⟩
+    simp [*]
+  tfae_have 2 → 3
+  · simp_rw [card_eq_one]
+    rintro ⟨hts, a, ha⟩
+    refine ⟨a, (mem_sdiff.1 $ superset_of_eq ha $ mem_singleton_self _).2, ?_⟩
+    rw [insert_eq, ←ha, sdiff_union_of_subset hts]
+  tfae_finish
+
+lemma covby_iff_card_sdiff_eq_one : t ⋖ s ↔ t ⊆ s ∧ (s \ t).card = 1 := covby_tfae.out 0 1
+
```

Comment:
I'm not convinced this TFAE is useful; you could achieve the same thing with something along the lines of
```suggestion
lemma covby_iff_card_sdiff_eq_one : t ⋖ s ↔ t ⊆ s ∧ (s \ t).card = 1 := by
  rw [covby_iff_exists_insert]
  constructor 
  · rintro ⟨a, ha, rfl⟩
    simp [*]
  · simp_rw [card_eq_one]
    rintro ⟨hts, a, ha⟩
    refine ⟨a, (mem_sdiff.1 $ superset_of_eq ha $ mem_singleton_self _).2, ?_⟩
    rw [insert_eq, ←ha, sdiff_union_of_subset hts]
```

Reply:
But surely that TFAE is useful if only to say  "This is all the equivalences we have in the library"?

Reply:
I agree we don't have 1 or 2 here, but this TFAE could still be useful in other contexts where going between any of this four concepts is useful. Though we do seemingly have a canonical spelling for this, so perhaps just the `covby_iff_` lemmas suffice.

Reply:
I've dumped the TFAE for what it's worth.